Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -2671,6 +2671,64 @@ public AndConstraint<TAssertions> OnlyHaveUniqueItems(string because = "", param
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        /// <summary>
+        /// Asserts that a collection contains only items which meet
+        /// the criteria provided by the inspector.
+        /// </summary>
+        /// <param name="expected">
+        /// The element inspector, which inspects each element in turn.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+        /// </param>
+        /// <exception cref="ArgumentNullException"><paramref name="expected"/> is <c>null</c>.</exception>
+        public AndConstraint<TAssertions> AllSatisfy(Action<T> expected, string because = "", params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(expected, nameof(expected), "Cannot verify against a <null> inspector");
+
+            var success = Execute.Assertion
+                                 .BecauseOf(because, becauseArgs)
+                                 .WithExpectation(
+                                     "Expected {context:collection} to contain only items satisfying the inspector{reason}, ")
+                                 .Given(() => Subject)
+                                 .ForCondition(subject => subject is not null)
+                                 .FailWith("but collection is <null>.")
+                                 .Then
+                                 .ForCondition(subject => subject.Any())
+                                 .FailWith("but collection is empty.")
+                                 .Then
+                                 .ClearExpectation();
+
+            if (success != false) return new AndConstraint<TAssertions>((TAssertions)this);
```

Comment:
Is this the intended way to check for failures? `success` is returning null in this case and I wasn't able to find another case where the result was checked in this way